aiohue/v2/controllers/events.py

```
@@ -0,0 +1,191 @@
+            else:
+                callback(type, data)
+
+    async def __event_reader(self) -> NoReturn:
+        """
+
+        Read incoming SSE messages and put them in a Queue to be processed.
+
+        Background task that keeps (re)connecting untill stopped.
+
+        https://html.spec.whatwg.org/multipage/server-sent-events.html#event-stream-interpretation
+        """
+        self._status = EventStreamStatus.CONNECTING
+        headers = {"Accept": "text/event-stream", "Cache-Control": "no-cache"}
+        retries = 0
+        while True:
+            if self._last_event_id:
+                headers["Last-Event-ID"] = self._last_event_id
+            try:
+                async with self._bridge.create_request(
+                    "get", "eventstream/clip/v2", timeout=0, headers=headers
+                ) as resp:
+                    # update status to connected once we reach this point
+                    self._status = EventStreamStatus.CONNECTED
+                    retries = 0  # reset on succesfull connect
+                    self._logger.debug("Connected to EventStream")
+                    # messages come in one by line, according to EventStream/SSE specs
+                    # we iterate over the incoming lines in the streamreader.
+                    # To prevent a deadlock waiting for a message while the connection is dead
+                    # we have a simple timeout guard in place.
+                    # If no message is received in 30 minutes, a Timeout error will be raised
+                    # and thus the connection re-established.
+                    iterator = resp.content.__aiter__()
+                    while True:
+                        line = await asyncio.wait_for(
+                            iterator.__anext__(), timeout=30 * 60
+                        )
+                        self.__parse_message(line)
+            except (ClientConnectionError, asyncio.TimeoutError) as err:
```

What about other client errors ?

---

Well, all connection related issued should be catched by the ClientConnectionError
If something else happens, it will be very bad and we wouldn't want to swallow the exception, right ?

Or can you think of something else I didn't think of ?